pkg/api/v1/conversion.go

```diff
@@ -381,6 +391,43 @@ func convert_v1_PodSpec_To_api_PodSpec(in *PodSpec, out *api.PodSpec, s conversi
 	} else {
 		out.ImagePullSecrets = nil
 	}
+
+	return nil
+}
+
+func convert_api_Pod_To_v1_Pod(in *api.Pod, out *Pod, s conversion.Scope) error {
+	if err := autoconvert_api_Pod_To_v1_Pod(in, out, s); err != nil {
+		return err
+	}
+	// We need to reset certain fields for mirror pods from pre-v1.1 kubelet
+	// (#15960).
+	// TODO: Remove this code after we drop support v1.0 kubelets.
+	if value, ok := in.Annotations[mirrorAnnotationKey]; ok && value == mirrorAnnotationValue {
+		// Reset the TerminationGracePeriodSeconds.
+		out.Spec.TerminationGracePeriodSeconds = nil
+		// Reset the resource requests.
+		for i := range out.Spec.Containers {
+			out.Spec.Containers[i].Resources.Requests = nil
+		}
+	}
+	return nil
+}
+
+func convert_v1_Pod_To_api_Pod(in *Pod, out *api.Pod, s conversion.Scope) error {
```

Comment:
I agree.

On Thu, Oct 22, 2015 at 12:34 PM, Yu-Ju Hong <notifications@github.com>
wrote:

> In pkg/api/v1/conversion.go
> <https://github.com/kubernetes/kubernetes/pull/16080#discussion_r42795288>
> :
>
> > +	}
> > +	// We need to reset certain fields for mirror pods from pre-v1.1 kubelet
> > +	// (#15960).
> > +	// TODO: Remove this code after we drop support v1.0 kubelets.
> > +	if value, ok := in.Annotations[mirrorAnnotationKey]; ok && value == mirrorAnnotationValue {
> > +		// Reset the TerminationGracePeriodSeconds.
> > +		out.Spec.TerminationGracePeriodSeconds = nil
> > +		// Reset the resource requests.
> > +		for i := range out.Spec.Containers {
> > +			out.Spec.Containers[i].Resources.Requests = nil
> > +		}
> > +	}
> > +	return nil
> > +}
> > +
> > +func convert_v1_Pod_To_api_Pod(in *Pod, out *api.Pod, s conversion.Scope) error {
>
> @lavalamp <https://github.com/lavalamp>, I don't think we need to muck
> with the conversion on this end since we will always reset the fields when
> converting to the versioned pod. Thoughts?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/16080/files#r42795288>.
>